examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py

```diff
@@ -121,12 +170,96 @@ def timestamp_from_airflow_date(airflow_date: str) -> float:
 
 @record
 class DagInfo:
+    webserver_url: str
     dag_id: str
     metadata: Dict[str, Any]
 
+    @property
+    def url(self) -> str:
+        return f"{self.webserver_url}/dags/{self.dag_id}"
+
+    @property
+    def dag_asset_key(self) -> AssetKey:
+        # Conventional asset key representing a successful run of an airfow dag.
+        return AssetKey(["airflow_instance", "dag", self.dag_id])
+
 
 @record
 class TaskInfo:
+    webserver_url: str
     dag_id: str
     task_id: str
     metadata: Dict[str, Any]
+
+    @property
+    def dag_url(self) -> str:
+        return f"{self.webserver_url}/dags/{self.dag_id}"
+
+
+@record
+class TaskInstance:
+    webserver_url: str
+    dag_id: str
+    task_id: str
+    run_id: str
+    metadata: Dict[str, Any]
+
+    @property
+    def note(self) -> str:
+        return self.metadata.get("note") or ""
+
+    @property
+    def details_url(self) -> str:
+        return f"{self.webserver_url}/dags/{self.dag_id}/grid?dag_run_id={self.run_id}&task_id={self.task_id}"
+
+    @property
+    def log_url(self) -> str:
+        return f"{self.details_url}&tab=logs"
+
+    @property
+    def start_date(self) -> float:
+        return AirflowInstance.timestamp_from_airflow_date(self.metadata["start_date"])
+
+    @property
+    def end_date(self) -> float:
+        return AirflowInstance.timestamp_from_airflow_date(self.metadata["end_date"])
```

Comment:
At some point we might cross the threshold of wanting to use an openapi-generated client https://github.com/apache/airflow-client-python